includes/User.php

```diff
@@ -2683,11 +2668,10 @@ public function setGlobalAttribute($attribute, $value) {
 		if ( $this->isPublicAttribute( $attribute ) ) {
 			$value = $this->replaceNewlineAndCRWithSpace( $value );
 			$this->userAttributes()->setAttribute( $this->getId(), new Attribute( $attribute, $value ) );
+		} else {
+			$this->setOptionHelper( $attribute, $value );
```

Comment:
That's correct. Any attribute stored in that attribute service is available publicly, meaning any client can request them. The private attributes are values that we don't want to expose to the outside world, like emails. Eventually we'll update the attribute service to support these and only return them to specific clients, but for now we'll just continue writing them to the user_properties table and let MW handle them.